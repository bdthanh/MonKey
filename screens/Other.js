import React, {useEffect, useState} from 'react';
import { View, Text, StyleSheet, ScrollView, TouchableOpacity, Image, Platform, Alert } from 'react-native';
import PressableText from '../components/Containers/PressableText';
import { getUserID, handleSignOut } from '../api/authentication';
import { StatusBarHeight } from '../components/constants';
import * as ImagePicker from 'expo-image-picker';
import { colors } from '../components/colors';
import { Avatar, Drawer } from 'react-native-paper';
import { copyDefaultCategory } from '../api/authentication';
import IncomeCategory from '../CategoriesList/IncomeCategory';
import { addDoc, collection } from 'firebase/firestore';
import { db } from '../api/db';
const {beige, brown, darkBlue, lightBlue, darkYellow, lightYellow} = colors;
const Other = ({navigation}) => {
  const [image, setImage] = useState('');
  const [username, setUsername] = useState('Team Grape')
  const pickImage = async () => {
    // No permissions request is necessary for launching the image library
    let result = await ImagePicker.launchImageLibraryAsync({
      mediaTypes: ImagePicker.MediaTypeOptions.Images,
      allowsEditing: true,
      aspect: [4, 4],
      quality: 1,
    });

    console.log(result);

    if (!result.cancelled) {
      setImage(result.uri);
    }
  };
  const askChangeAvatar = () => {
    Alert.alert("Change avatar?", "", [
      {text: 'No', onPress: () => console.log('Alert closed')},
      {text: 'Yes', onPress: pickImage}
    ]);
  }

  return (
    <>
      <View style={styles.header}>
        <Text style={styles.boldBlueHeaderText}>Other</Text>
      </View>
      <ScrollView>
        <View style={styles.userView}>
          <View style={styles.imageView}>
            <TouchableOpacity onPress={askChangeAvatar}>
            {image == '' && <Avatar.Image source={{ uri: "https://cdn.landesa.org/wp-content/uploads/default-user-image.png" }} size={75} />}
            {image != '' && <Avatar.Image source={{ uri: image }} size={75} />}
            </TouchableOpacity>
          </View>
          <View style={styles.usernameView}>
            <Text style={styles.boldBlackHeaderText}>{username}</Text>
          </View>
        </View>
        <PressableText onPress={() => {
          handleSignOut(navigation)
        }}>
          Sign Out
        </PressableText>
        {/* <PressableText onPress={() => {
          const IncomeDefault = collection(db, 'Input Category/Income/default')
          IncomeCategory.forEach((doc) => {if (doc.name != 'Edit') addDoc(IncomeDefault, {
            name: doc.name,
            icon: doc.icon,
            color: doc.color
          })})
        }}>
          copyDefaultCategory
        </PressableText> */}
      </ScrollView>
    </>
  );
}

const styles = StyleSheet.create({
  mainContainer: {
    flex: 1,
    padding: 25,
    paddingTop: StatusBarHeight + 30,
    backgroundColor: beige,
  }, 
  boldBlueHeaderText: {
    fontSize: 34,
    color: darkBlue,
    marginBottom: 10,
    fontWeight: 'bold',
  },
  boldBlackHeaderText: {
    fontSize: 25,
    fontWeight: '500',
    color: '#000',
    marginBottom: 10,
  },
  header: {
    alignItems:'center', 
    justifyContent:'flex-end',
    backgroundColor:lightYellow,
    borderBottomColor:'#808080',
    borderBottomWidth:1,
    paddingTop:3,
    height: StatusBarHeight + 48,
  },
  userView: {
    flexDirection: 'row',
    height:120,
    paddingLeft:20
  },
  imageView: {
    flex: 3,
    justifyContent:'center'
  },
  usernameView: {
    flex: 7,
    paddingLeft:0,
    alignItems:'flex-start',
    justifyContent:'center',
  }
})

export default Other;