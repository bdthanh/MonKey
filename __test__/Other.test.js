import React from 'react';
import Other from '../screens/Other';
import { NavigationContainer } from '@react-navigation/native';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const tree = renderer
    .create(
      <NavigationContainer>
        <Other/>
      </NavigationContainer>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});