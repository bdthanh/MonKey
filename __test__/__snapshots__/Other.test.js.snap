// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#cce9f6",
        "borderBottomColor": "#808080",
        "borderBottomLeftRadius": 20,
        "borderBottomRightRadius": 20,
        "borderBottomWidth": 0,
        "height": 62,
        "justifyContent": "flex-end",
        "paddingTop": 3,
        "shadowColor": "#35baf6",
        "shadowOffset": Object {
          "height": 1,
          "width": 0,
        },
        "shadowOpacity": 0.8,
        "shadowRadius": 3,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#35baf6",
          "fontSize": 34,
          "fontWeight": "bold",
          "marginBottom": 10,
        }
      }
    >
      Other
    </Text>
  </View>
  <RCTScrollView
    style={
      Object {
        "backgroundColor": "#f3f3f3",
        "flex": 1,
        "marginTop": 6,
      }
    }
  >
    <View>
      <View
        style={
          Object {
            "backgroundColor": "#fff",
            "borderBottomColor": "#E9E9E9",
            "borderBottomWidth": 1,
            "flexDirection": "row",
            "paddingBottom": 15,
            "paddingLeft": 20,
            "paddingTop": 12,
          }
        }
      >
        <View
          style={
            Object {
              "flex": 3,
              "justifyContent": "center",
            }
          }
        >
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Array [
                  Object {
                    "backgroundColor": "#6200ee",
                    "borderRadius": 50,
                    "height": 100,
                    "width": 100,
                  },
                  undefined,
                ]
              }
            >
              <Image
                source={
                  Object {
                    "uri": "https://cdn.landesa.org/wp-content/uploads/default-user-image.png",
                  }
                }
                style={
                  Object {
                    "borderRadius": 50,
                    "height": 100,
                    "width": 100,
                  }
                }
              />
            </View>
          </View>
        </View>
        <View
          style={
            Object {
              "alignItems": "flex-start",
              "flex": 7,
              "justifyContent": "center",
              "paddingLeft": 0,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#FAB00E",
                "fontSize": 24,
                "fontWeight": "500",
              }
            }
          >
            Good night!
          </Text>
          <Text
            style={
              Object {
                "color": "#494949",
                "fontSize": 28,
                "fontWeight": "500",
              }
            }
          >
            
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#f3f3f3",
            "borderTopColor": "#E9E9E9",
            "borderTopWidth": 1,
            "flexDirection": "row",
            "height": 36,
            "justifyContent": "center",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#b2b2b2",
              "fontSize": 16,
            }
          }
        >
          Personalize
        </Text>
      </View>
      <View
        style={
          Object {
            "backgroundColor": "#fff",
            "borderTopColor": "#E9E9E9",
            "borderTopWidth": 1,
            "height": 48,
            "justifyContent": "center",
            "paddingBottom": 4,
            "paddingLeft": 12,
            "paddingRight": 16,
            "paddingTop": 4,
          }
        }
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Object {
                "flexDirection": "row",
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flex": 8,
                  "flexDirection": "row",
                  "justifyContent": "flex-start",
                }
              }
            >
              <Text />
              <Text
                style={
                  Object {
                    "color": "#494949",
                    "fontSize": 16,
                  }
                }
              >
                  Edit username
              </Text>
            </View>
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flex": 2,
                  "flexDirection": "row",
                  "justifyContent": "flex-end",
                }
              }
            >
              <Text />
            </View>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "backgroundColor": "#fff",
            "borderTopColor": "#E9E9E9",
            "borderTopWidth": 1,
            "height": 48,
            "justifyContent": "center",
            "paddingBottom": 4,
            "paddingLeft": 12,
            "paddingRight": 16,
            "paddingTop": 4,
          }
        }
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Object {
                "flexDirection": "row",
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flex": 8,
                  "flexDirection": "row",
                  "justifyContent": "flex-start",
                }
              }
            >
              <Text />
              <Text
                style={
                  Object {
                    "color": "#494949",
                    "fontSize": 16,
                  }
                }
              >
                  Expense limit setting
              </Text>
            </View>
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flex": 2,
                  "flexDirection": "row",
                  "justifyContent": "flex-end",
                }
              }
            >
              <Text />
            </View>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "backgroundColor": "#fff",
            "borderTopColor": "#E9E9E9",
            "borderTopWidth": 1,
            "height": 48,
            "justifyContent": "center",
            "paddingBottom": 4,
            "paddingLeft": 12,
            "paddingRight": 16,
            "paddingTop": 4,
          }
        }
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Object {
                "flexDirection": "row",
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flex": 8,
                  "flexDirection": "row",
                  "justifyContent": "flex-start",
                }
              }
            >
              <Text />
              <Text
                style={
                  Object {
                    "color": "#494949",
                    "fontSize": 16,
                  }
                }
              >
                  Income category setting
              </Text>
            </View>
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flex": 2,
                  "flexDirection": "row",
                  "justifyContent": "flex-end",
                }
              }
            >
              <Text />
            </View>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "backgroundColor": "#fff",
            "borderTopColor": "#E9E9E9",
            "borderTopWidth": 1,
            "height": 48,
            "justifyContent": "center",
            "paddingBottom": 4,
            "paddingLeft": 12,
            "paddingRight": 16,
            "paddingTop": 4,
          }
        }
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Object {
                "flexDirection": "row",
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flex": 8,
                  "flexDirection": "row",
                  "justifyContent": "flex-start",
                }
              }
            >
              <Text />
              <Text
                style={
                  Object {
                    "color": "#494949",
                    "fontSize": 16,
                  }
                }
              >
                  Expense category setting
              </Text>
            </View>
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flex": 2,
                  "flexDirection": "row",
                  "justifyContent": "flex-end",
                }
              }
            >
              <Text />
            </View>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#f3f3f3",
            "borderTopColor": "#E9E9E9",
            "borderTopWidth": 1,
            "flexDirection": "row",
            "height": 36,
            "justifyContent": "center",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#b2b2b2",
              "fontSize": 16,
            }
          }
        >
          MonKey
        </Text>
      </View>
      <View
        style={
          Object {
            "backgroundColor": "#fff",
            "borderTopColor": "#E9E9E9",
            "borderTopWidth": 1,
            "height": 48,
            "justifyContent": "center",
            "paddingBottom": 4,
            "paddingLeft": 12,
            "paddingRight": 16,
            "paddingTop": 4,
          }
        }
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Object {
                "flexDirection": "row",
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flex": 8,
                  "flexDirection": "row",
                  "justifyContent": "flex-start",
                }
              }
            >
              <Text />
              <Text
                style={
                  Object {
                    "color": "#494949",
                    "fontSize": 16,
                  }
                }
              >
                  Help
              </Text>
            </View>
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flex": 2,
                  "flexDirection": "row",
                  "justifyContent": "flex-end",
                }
              }
            >
              <Text />
            </View>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "backgroundColor": "#fff",
            "borderTopColor": "#E9E9E9",
            "borderTopWidth": 1,
            "height": 48,
            "justifyContent": "center",
            "paddingBottom": 4,
            "paddingLeft": 12,
            "paddingRight": 16,
            "paddingTop": 4,
          }
        }
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Object {
                "flexDirection": "row",
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flex": 8,
                  "flexDirection": "row",
                  "justifyContent": "flex-start",
                }
              }
            >
              <Text />
              <Text
                style={
                  Object {
                    "color": "#494949",
                    "fontSize": 16,
                  }
                }
              >
                  About us
              </Text>
            </View>
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flex": 2,
                  "flexDirection": "row",
                  "justifyContent": "flex-end",
                }
              }
            >
              <Text />
            </View>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#f3f3f3",
            "borderTopColor": "#E9E9E9",
            "borderTopWidth": 1,
            "flexDirection": "row",
            "height": 36,
            "justifyContent": "center",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#b2b2b2",
              "fontSize": 16,
            }
          }
        >
          Feedback
        </Text>
      </View>
      <View
        style={
          Object {
            "backgroundColor": "#fff",
            "borderTopColor": "#E9E9E9",
            "borderTopWidth": 1,
            "height": 48,
            "justifyContent": "center",
            "paddingBottom": 4,
            "paddingLeft": 12,
            "paddingRight": 16,
            "paddingTop": 4,
          }
        }
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Object {
                "flexDirection": "row",
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flex": 8,
                  "flexDirection": "row",
                  "justifyContent": "flex-start",
                }
              }
            >
              <Text />
              <Text
                style={
                  Object {
                    "color": "#494949",
                    "fontSize": 16,
                  }
                }
              >
                  Rate this MonKey
              </Text>
            </View>
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flex": 2,
                  "flexDirection": "row",
                  "justifyContent": "flex-end",
                }
              }
            >
              <Text />
            </View>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#f3f3f3",
            "borderTopColor": "#E9E9E9",
            "borderTopWidth": 1,
            "flexDirection": "row",
            "height": 36,
            "justifyContent": "center",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#b2b2b2",
              "fontSize": 16,
            }
          }
        >
          Sign out
        </Text>
      </View>
      <View
        style={
          Object {
            "backgroundColor": "#fff",
            "borderBottomColor": "#E9E9E9",
            "borderBottomWidth": 1,
            "borderTopColor": "#E9E9E9",
            "borderTopWidth": 1,
            "height": 48,
            "justifyContent": "center",
            "paddingBottom": 4,
            "paddingLeft": 12,
            "paddingRight": 16,
            "paddingTop": 4,
          }
        }
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Object {
                "flexDirection": "row",
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "flex-start",
                }
              }
            >
              <Text />
              <Text
                style={
                  Object {
                    "color": "#494949",
                    "fontSize": 16,
                    "fontWeight": "bold",
                  }
                }
              >
                  Sign out
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
